Automattic/wp-calypso
# Add delete button for domain registrations and mappings

#### Changes proposed in this Pull Request

This PR adds a delete button for on the registered domain and mapped domain screens, so that the user doesn't have to go to the purchases list.

Here's how it appears in the registered domain screen:

<img width="732" alt="Screenshot 2020-03-18 at 6 18 23 PM" src="https://user-images.githubusercontent.com/13062352/76988448-feb21e00-6944-11ea-9212-0e604ea36b2f.png">

Here's how it appears in the mapped domain screen:

<img width="733" alt="domainnnnnnnn" src="https://user-images.githubusercontent.com/13062352/77168261-d434a200-6ab7-11ea-8314-29988496623f.png">

#### Testing instructions

* Ensure the delete button appears like in the screenshots above, for both mapped domains and registered domains
* For non-refundable registered or mapped domains, make sure the delete button will show the respective dialog
* For refundable domains, make sure the button sends you to the appropriate cancel purchase view
